hw/xbox/nv2a/pgraph_texture_replacer.c

```diff
@@ -0,0 +1,26 @@
+#include "pgraph_texture_replacer.h"
+
+
+void texture_download(TextureBinding *texture, SurfaceBinding *surface)
+{
+    void *pixels = surface->texture;
```

Comment:
To avoid the confusion with the double free, don't create a local variable for this, just use surface->texture directly.